pyteal/compiler/compiler.py

```diff
@@ -209,7 +212,7 @@ def compileSubroutine(
     newSubroutines = referencedSubroutines - subroutine_start_blocks.keys()
     for subroutine in sorted(newSubroutines, key=lambda subroutine: subroutine.id):
         compileSubroutine(
-            subroutine.get_declaration(),
+            subroutine.get_declaration_by_version(options.version),
```

Comment:
Comment relating to this file in general, not this line: I had a verbal discussion with @michaeldiamant and we thought it would be a good idea to add some way of opting out of frame pointers for v8 programs.

The reason being that we've already shipped a pyteal version capable of generating v8 programs _without_ frame pointers. Because frame pointer usages _does_ affect the runtime behavior of programs in some circumstances, it seems like a good idea to give users the option to opt-out of them, since we have to support both cases anyway.

There are a few ways this could be implemented, but I imagine we could add a boolean `frame_pointers` field to `CompileOptions` which would default to true if the version is 8 or higher, and false otherwise.

Reply:
The suggested change has been pushed in this commit: https://github.com/algorand/pyteal/pull/562/commits/4b274508d3c47e48fa0700642e2d7a25d7a246b2

Reply:
I think we might want the corresponding integration tests?